<?xml version="1.0"?>
<launch>
	<arg name="mode" default="$(env MOVEL_MODE)" doc="mode: [deployment, development]"/>
	<!-- room exploration server node -->
	<!--remap from="/room_exploration_server/path" to="/path_saver/path"/ -->
	<group if="$(eval mode == 'deployment')">
		<node pkg="ipa_room_exploration" type="room_exploration_server" name="room_exploration_server" output="screen" respawn="true" respawn_delay="2">
			<rosparam command="load" file="$(find ipa_room_exploration)/ros/launch/room_exploration_action_server_params.yaml"/>
			<rosparam command="load" file="$(env HOME)/.config/movel/config/coverage_path_planner/config/room_exploration_action_server_params.yaml"/>
			<param name="file_name" value="coverage_path"/>
		</node>
	</group>

	<group unless="$(eval mode == 'deployment')">
		<node pkg="ipa_room_exploration" type="room_exploration_server" name="room_exploration_server" output="screen" respawn="true" respawn_delay="2">
			<rosparam command="load" file="$(find ipa_room_exploration)/ros/launch/room_exploration_action_server_params.yaml"/>
			<param name="file_name" value="coverage_path"/>
		</node>
	</group>

	<!-- include file="$(find ipa_room_exploration)/ros/test/room_exploration_client.launch"/ -->
	<!-- ACTIVATE IF NEEDED: run cob_map_accessibility_analysis_server -->
	<!-- include file="$(find ipa_room_exploration)/ros/launch/cob_map_accessibility_analysis.launch"/-->

	<!-- NOT NECESSARY ANYMORE: using direct library interface now, run coverage_check_server -->
	<!-- include file="$(find ipa_room_exploration)/ros/launch/coverage_check_server.launch"/-->

</launch>
